### PR TITLE
Handle structured config file

### PR DESCRIPTION
**Pull request by geyslan**
Close: #3182 

### 1. Explain what the PR does

d3f9ba9fb **chore(config): add gotemplate and content-type**
86c120975 **chore!: change crs flag/key to cri**
d1532749d **feat(docs): update docs for structured config file**
9092b9941 **feat(cobra): add parser for structured config file**

d3f9ba9fb **chore(config): add gotemplate and content-type**

```
Since #3622, webhooks can be configured with a gotemplate and a
content-type. This commit adds these two fields to the config file.
```

86c120975 **chore!: change crs flag/key to cri**

```
BREAKING CHANGE: change crs flag to cri
```

d1532749d **feat(docs): update docs for structured config file**

```
This also changes the default config file to be a structured YAML file
and adds a new example config file 'global_config_cli.yaml'  that uses
the CLI format.
```

9092b9941 **feat(cobra): add parser for structured config file**

```
This commit adds a parser for a structured config file using viper.

Now, the config file can be written in a more structured way, like:

cache:
    type: mem
    size: 1024

The old way (cli flags) is still supported.

cache:
    - cache-type=mem
    - mem-cache-size=556
```

### 2. Explain how to test it

`sudo ./dist/tracee --config ./examples/config/global_config.yaml`
`sudo ./dist/tracee --config ./examples/config/global_config_cli.yaml`

### CODE REVIEW

**Comment by geyslan** [comment]
This doesn't inject the new configuration structures to be used later by flag validations. That would be interesting, but the effort would be much greater. In any case, these structures take care of outputting the flags - `CLIFlagger.flags()` - in the format they need to be parsed (prepared).

File: examples/config/global_config.yaml [hunk 1]
```diff
@@ -1,21 +1,136 @@
...
+    bypass: false
+    # add:
+    #     - cap_sys_admin
+    #     - cap_syslog
+    # drop:
+    #     - cap_chown
+
 containers: false
-crs: []
+crs:
```
Comment (yanivagman) [inline examples/config/global_config.yaml:21]: We should rename this to cri, which is much more clear (can be in a different PR)

**Review by yanivagman — COMMENTED** [review]
Looks good

File: examples/config/global_config.yaml (hunk 1 above)
Comment (geyslan) [reply examples/config/global_config.yaml]: Done: 33686e61ac786002789b4ae560c8bd685f14e972

**Review by rafaeldtinoco — APPROVED** [review]
I have to check Lunar 6.2 test for hooked_syscall, so Ignore that error. Loved the amount of tests, for each specific option. Job well done. LGTM.
